### PR TITLE
feat: re-render expanded columns when the columns change

### PR DESCRIPTION
**Pull request by 0ces**
With this we will re-render the expanded columns whenever the columns change

### CODE REVIEW

**Review by 0ces — COMMENTED** [review]
@bdow should we keep the `refetchExpandedColumns`? I'd say no as with this it will be automatically made

File: src/components/Table/utils/TableState.ts [hunk 1]
```diff
@@ -228,16 +228,16 @@ export class TableState {
       // (We'll add to this after figuring out which columns are new vs existing.)
       const columnIdsToExpand = isInitial ? localStorageColumns : [];
```
Comment (bdow) [inline src/components/Table/utils/TableState.ts:229]: **question:** I was thinking that `columnIdsToExpand` would initialize with `this.expandedColumnIds` if `!isInitial`. Would that work here?

**Comment by bdow** [comment]
> @bdow should we keep the `refetchExpandedColumns`? I'd say no as with this it will be automatically made

Yea, I'm okay with removing this.

File: src/components/Table/utils/TableState.ts (hunk 1 above)
Comment (0ces) [reply src/components/Table/utils/TableState.ts]: Yes, it does work

File: src/components/Table/utils/TableState.ts [hunk 2]
```diff
@@ -226,15 +226,13 @@ export class TableState {
       // On initial render the columns we start with is whatever is in local storage.
       // On any subsequent render, we start with an empty array.
       // (We'll add to this after figuring out which columns are new vs existing.)
-      const columnIdsToExpand = isInitial ? localStorageColumns : [];
-
-      // Create a list of all existing column ids. (We ignore the `initExpanded` property for existing columns)
-      const existingColumnIds = this.columns.map((c) => c.id);
```
Comment (bdow) [inline src/components/Table/utils/TableState.ts:232]: **question:** Is there a reason why we are no longer ignoring the `initExpanded` for existing columns? The idea behind this was that if a user collapsed a column, then the columns rerender we wouldn't want to expand it s the user explicitly collapsed it.

Comment (0ces) [reply src/components/Table/utils/TableState.ts]: No, there is no reason for that change, I'll leave it as it was

**Review by bdow — APPROVED** [review]

**Pull request merged**